### PR TITLE
dist/tools/define2u16: provide tool to create somewhat unique define values

### PR DESCRIPTION
**Pull request by kaspar030**
### Contribution description

Seeing people getting tired of `#define FOO 12345`, I tried to come up with something that creates somewhat unique values that can be used for e.g., message IDs.

This PR adds a toool that basically prints `crc32(str)`.
Additionally, there's a minimum value (currently 100001), and if the result would be below that, it would re-do `crc32(<value><str>)`, until the result is >= 100001. This is to keep some of the number space for human-scale numbers (e.g., enums starting low, ...).

Another commit changes `MSG_XTIMER` to the output of the tool.

What do you think?

### CODE REVIEW

File: dist/tools/define2u32/define2u32.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# Copyright (c) 2020 Kaspar Schleiser <kaspar@schleiser.de>
+#                    Freie Universität Berlin
+#                    Inria
+#
+# This utiliy turns a string into it's crc32 as hex value.
+#
+# If the resulting value is smaller than MIN_VALUE, the concatination of the
+# value as hex string and the original string will be crc32()ed again, until
+# `res > MIN_VALUE`.
+#
```
Comment (bergzand) [inline dist/tools/define2u32/define2u32.py]: Maybe add a short usage example here:

```
# Usage example:
#
# $ ./define2u32.py MSG_XTIMER
# #define MSG_XTIMER 0x8eafaa52 /* created with dist/tools/define2u32.py */
```

**Comment by kaspar030** [comment]
@miri64 just pointed out that the msg type field is only 16bit...

**Comment by bergzand** [comment]
> @miri64 just pointed out that the msg type field is only 16bit...

So `define2u16.py` then? With CRC16?

**Comment by kaspar030** [comment]
> So `define2u16.py` then? With CRC16?

yup. I'm folding the upper and lower 16bit of crc32 using xor. The algorithm doesn't really matter, I guess.

I've set the minimum value to 0xa000, maybe that's too high?

**Comment by miri64** [comment]
Sorry for coming in late: why not provide both?

```
./new_define.py -h
-w int   (defaults to 16)
NAME   name for define
```

**Comment by miri64** [comment]
Makes it reusable for other use cases. ;-)

**Comment by kaspar030** [comment]
> Sorry for coming in late: why not provide both?

will do.

**Comment by miri64** [comment]
Ping @kaspar030?

**Comment by fjmolinas** [comment]
ping @kaspar030!

**Pull request closed without merging**